appveyor.yml

```
@@ -34,87 +34,93 @@ install:
+  - ps: if($env:APPVEYOR_PULL_REQUEST_NUMBER) { $Params += "/d:sonar.pullrequest.key=$env:APPVEYOR_PULL_REQUEST_NUMBER", "/d:sonar.pullrequest.branch=`"$env:APPVEYOR_REPO_BRANCH`"" }
+  - ps: Start-process "dotnet" "sonarscanner begin $($Params -join ' ')"
+  - codeclimate-test-reporter before-build
+  - dotnet build --verbosity minimal %SOLUTION_NAME%.sln
+  - ps: $TEST_PROJECTS = (Get-ChildItem -Path .\Tests\**\ -Recurse -Include *.csproj).Fullname
+  - ps: |
+      foreach($testProject in $TEST_PROJECTS)
+      {
+          dotnet test $testProject --no-build --verbosity minimal /p:CollectCoverage=true "/p:CoverletOutputFormat=\`"cobertura,opencover,lcov\`"" --logger:"junit;LogFilePath=test-results.xml"
+          $path = Split-Path $testProject
+          $wc = New-Object 'System.Net.WebClient'
+          $wc.UploadFile("https://ci.appveyor.com/api/testresults/xunit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path $path/test-results.xml))          
```

yamllint

[error] trailing spaces

(trailing-spaces)